pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/MessageDeduplication.java

```diff
@@ -323,6 +326,19 @@ public MessageDupStatus isDuplicate(PublishContext publishContext, ByteBuf heade
 
         String producerName = publishContext.getProducerName();
         long sequenceId = publishContext.getSequenceId();
+        // The process of the Producer sending chunk messages is continuous, and all chunks of a message use the same
+        // message metadata and sequence ID. Therefore, it is only necessary to check if the sequence ID of the first
+        // chunk is duplicated.
+        // When we receive the initial message of a non-duplicated chunk message, we place it in the
+        // chunkMessageOngoing. Upon completion of sending this chunk message, if we receive other messages
+        // sent by this Producer, we will remove it from the chunkMessageOngoing.
+        if (chunkMessageOngoing.containsKey(producerName)) {
+            if (publishContext.isChunked() && chunkMessageOngoing.get(producerName).equals(sequenceId)) {
+                return MessageDupStatus.NotDup;
```

Comment:
How can we dedup if the same chunk is sent repeatedly?

For example,
m1-chunk-1, seqid=1
m1-chunk-2, seqid=1
m1-chunk-2, seqid=1 //don't we need to dedup this?
m1-chunk-3, seqid=1
m2-chunk-1, seqid=2

Reply:
Fixed.